src/terms/operators.jl

```diff
@@ -133,7 +133,7 @@ Matrix(op::MagneticFieldOperator) = error("Not implemented")
 function optimize_operators_(ops)
     RSmults = [op for op in ops if op isa RealSpaceMultiplication]
     isempty(RSmults) && return ops
-    nonRSmults = [op for op in ops if !(op isa RealSpaceMultiplication)]
+    nonRSmults = [op for op in ops if !(op isa RealSpaceMultiplication) && !(op isa NoopOperator)]
```

Comment:
I think the logic is clearer if you just do this as the first line of the function:
```
ops = filter(op -> !(op isa NoopOperator), ops)
```

Reply:
good idea